packages/blade/src/components/Dropdown/DropdownOverlay.web.tsx

```diff
@@ -215,78 +84,21 @@ const _DropdownOverlay = ({ children, testID }: DropdownOverlayProps): React.Rea
     // eslint-disable-next-line react-hooks/exhaustive-deps
   }, [isOpen]);
 
-  // We want to set width of overlay as per width of the SelectInput
-  React.useEffect(() => {
-    // We don't set width according to trigger when trigger is Button or other menu trigger
-    if (isMenu) {
-      return undefined;
-    }
-
-    const setOverlayWidth = throttle((): void => {
-      if (triggererRef.current?.clientWidth && hasLabelOnLeft) {
-        const svgWidth: number = size[16];
-        const interactionElementPadding: number = theme.spacing[4];
-        const offset = svgWidth + interactionElementPadding;
-        // SelectInput is -> Button + InteractionElement on right (the chevron icon)
-        // So we add the interactionElement offset with Button's width.
-        setWidth(makeSize(triggererRef.current?.clientWidth + offset));
-      } else {
-        // We don't have to worry about setting the custom width when label is on top since we can just 100% width of parent div
-        setWidth('100%');
-        window.removeEventListener('resize', setOverlayWidth);
-      }
-    }, 1000);
-
-    setOverlayWidth();
-    window.addEventListener('resize', setOverlayWidth);
-    return () => {
-      window.removeEventListener('resize', setOverlayWidth);
-    };
-    // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, [setWidth, triggererRef, hasLabelOnLeft]);
-
-  const onAnimationEnd = React.useCallback(() => {
-    if (isOpen) {
-      setShowFadeOutAnimation(true);
-    } else {
-      setShowFadeOutAnimation(false);
-    }
-  }, [isOpen]);
-
-  const styles = React.useMemo(() => ({ opacity: isOpen ? 1 : 0 }), [isOpen]);
-
   return (
-    // eslint-disable-next-line @typescript-eslint/no-explicit-any
-    <BaseBox position="relative" ref={refs.setFloating as any}>
-      {isOpen ? (
-        <BaseBox
-          position={'fixed' as never}
-          top="spacing.0"
-          left="spacing.0"
-          height="100%"
-          width="100%"
-          onClick={() => setIsOpen(false)}
-        />
-      ) : null}
-      <AnimatedOverlay
+    <BaseBox
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
+      ref={refs.setFloating as any}
```

Comment:
maybe as `never` so you NEVER mess with it? 😆

Reply:
<img width="277" alt="image" src="https://github.com/razorpay/blade/assets/30949385/5bcdf8ca-a92e-48d5-af56-1e7fb9fb7d7f">

In most places where we use `any` actually. Can relook into it later if required.